### PR TITLE
Add Uniqueness Validation Message when new user added

### PR DESCRIPTION
**Pull request by Loop97**
When adding a not unique username or email. Will get notification and redirected to user page.

### CODE REVIEW

File: spec/factories/users.rb [hunk 1]
```diff
@@ -42,5 +42,6 @@
     email
     password { '123456' }
     cell { User::CELLS.sample }
+    username { (0...8).map { rand(65..90).chr }.join }
```
Comment (indocomsoft) [inline spec/factories/users.rb]: Why not `SecureRandom.alphanumeric(8)`

File: app/models/user.rb [hunk 2]
```diff
@@ -57,6 +57,7 @@ class User < ApplicationRecord
   has_many :availabilities, dependent: :destroy
   validates :cell, presence: true
   validates :email, presence: true
-
+  validates :email, uniqueness: true
```
Comment (indocomsoft) [inline app/models/user.rb]: Add tests for this in the model specs

File: app/controllers/registrations_controller.rb [hunk 3]
```diff
@@ -14,8 +14,15 @@ def create
...
     resource.save
+    success = resource.save
     yield resource if block_given?
-    redirect_to users_path
+    if success
+      redirect_to users_path,
+        notice:'New User Created'
+    else
+      redirect_to users_path,
+        alert: 'User name already exist!'
```
Comment (indocomsoft) [inline app/controllers/registrations_controller.rb]: Technically not necessarily true. Can instead just use `resource.errors.full_messages.join(', ')`

File: app/controllers/registrations_controller.rb [hunk 4]
```diff
@@ -14,8 +14,12 @@ def create
     ensure_admin || return
     build_resource(sign_up_params)
     resource.save
+    success = resource.save
     yield resource if block_given?
-    redirect_to users_path
+    return unless success
```
Comment (indocomsoft) [inline app/controllers/registrations_controller.rb]: This should be 
```ruby
if success
  redirect_to users_path, notice: "Created user successfully"
else
  redirect_to users_path, alert: resource.errors.full_messages.join(',')
end
```

File: app/models/user.rb (hunk 2 above)
Comment (indocomsoft) [reply app/models/user.rb]: Bump

Comment (Loop97) [reply app/models/user.rb]: would this be the same as the test case for Username { SecureRandom.alphanumeric(8) }? but appending the email suffix (e.x. @gmail.com)

File: app/models/user.rb [hunk 5]
```diff
@@ -57,6 +57,7 @@ class User < ApplicationRecord
   has_many :availabilities, dependent: :destroy
   validates :cell, presence: true
   validates :email, presence: true
```
Comment (indocomsoft) [inline app/models/user.rb]: Please combine the validations into one line:
`validates :email, presence: true, uniqueness: true`

**Pull request merged**